compiler/lib/link_js.ml

```diff
@@ -37,57 +39,61 @@ module Line_reader : sig
 
   val drop : t -> unit
 
-  val close : t -> unit
+  val reset : t -> unit
 
   val lnum : t -> int
 
   val fname : t -> string
 end = struct
   type t =
-    { ic : in_channel
-    ; fname : string
-    ; mutable next : string option
+    { fname : string
+    ; lines : string list
+    ; mutable current : string list
     ; mutable lnum : int
     }
 
-  let close t = close_in t.ic
+  let reset t =
+    t.current <- t.lines;
+    t.lnum <- 0
 
   let open_ fname =
-    let ic = open_in_bin fname in
-    { ic; lnum = 0; fname; next = None }
+    let lines =
+      (* If we are not on 32-bit hardware (where the max string length is too
+         small), read the entire file and split it in lines. This is faster
+         than reading it line by line. On 32 bits, we fall back to a
+         line-by-line read. *)
+      let ic = open_in_bin fname in
+      let x =
+        if Sys.word_size >= 64
+        then
+          really_input_string ic (in_channel_length ic) |> String.split_on_char ~sep:'\n'
```

Comment:
For the record, the two branches are not equivalent.
for empty content, `String.split_on_char` returns `[""]`, input_lines return `[]`.
for content with a trailing newline, `String.split_on_char` will return an extra empty line compared to `input_lines`.